include/nanorange/algorithm/unique.hpp

```diff
@@ -18,30 +19,30 @@ namespace detail {
 struct unique_fn {
 private:
     template <typename I, typename S, typename R, typename Proj>
-    static constexpr I impl(I first, S last, R& comp, Proj& proj)
+    static constexpr subrange<I> impl(I first, S last, R& comp, Proj& proj)
     {
-        first = adjacent_find_fn::impl(std::move(first), last, comp, proj);
+        I it = adjacent_find_fn::impl(first, last, comp, proj);
 
-        if (first == last) {
-            return first;
+        if (it == last) {
+            return {std::move(it), std::move(nano::next(first, last))};
         }
 
-        for (I n = next(first, 2, last); n != last; ++n) {
-            if (!nano::invoke(comp, nano::invoke(proj, *first),
+        for (I n = next(it, 2, last); n != last; ++n) {
+            if (!nano::invoke(comp, nano::invoke(proj, *it),
                               nano::invoke(proj, *n))) {
-                *++first = iter_move(n);
+                *++it = iter_move(n);
             }
         }
 
-        return ++first;
+        return {std::move(++it), std::move(nano::next(first, last))};
```

Comment:
We've already iterated to the end of the range in the `for` loop above -- move the declaration of `n` outside the loop, then you can `return {next(it), move(n)};`